go/ephemeral/user_ek_box_storage.go

```diff
@@ -0,0 +1,249 @@
+package ephemeral
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/keybase/client/go/libkb"
+	"github.com/keybase/client/go/protocol/keybase1"
+)
+
+type UserEKBoxMap map[keybase1.EkGeneration]keybase1.UserEkBoxed
+type UserEKMap map[keybase1.EkGeneration]keybase1.UserEk
+
+// We cache UserEKBoxes from the server in memory and a persist to a local
+// KVStore.
+type UserEKBoxStorage struct {
+	libkb.Contextified
+	sync.Mutex
+	indexed bool
+	cache   UserEKBoxMap
+}
+
+func NewUserEKBoxStorage(g *libkb.GlobalContext) *UserEKBoxStorage {
+	return &UserEKBoxStorage{
+		Contextified: libkb.NewContextified(g),
+		cache:        make(UserEKBoxMap),
+	}
+}
+
+func (s *UserEKBoxStorage) dbKey() libkb.DbKey {
+	key := fmt.Sprintf("user-ek-box-%s", s.G().Env.GetUsername())
```

Comment:
Should we include the eldest seqno in this key, to avoid wacky reset corner cases?

Reply:
what's the corner case?

Reply:
resolved offline and added!